frontend/src/AppRouter.re

```diff
@@ -0,0 +1,38 @@
+module ReactIntlProvider = {
+  [@react.component]
+  let make = (~locale, ~messages, ~children) => ReasonReact.element(ReactIntl.IntlProvider.make(~locale, ~messages, children));
+};
+
+[@react.component]
+let make = () => {
+  let url = ReasonReactRouter.useUrl();
+  let (language, setLanguage) = React.useState(() => Language.defaultLanguage);
+  let (keycloak, _) = Keycloak.useKeycloak();
+
+  let page =
+    keycloak |> Keycloak.authenticated
+      ? switch (url.path) {
+        | [] => <HomePage />
+        | ["dashboard"] => <Dashboard />
+        | ["profile"] => <UserProfile />
+        | ["new-game"] => <GameWizard />
+        | ["games"] => <GameList />
+        | ["games", gameId] => <GameDetails gameId />
+        | ["games", gameId, "new-match"] => <MatchWizard gameId />
+        | ["games", gameId, "matches"] => <MatchList gameId />
+        | ["games", gameId, "my-scripts"] => <MyScripts gameId />
+        | ["games", "matches", matchId] => <MatchDetails matchId />
+        | ["games", "matches", matchId, "new-script"] => <ScriptWizard matchId />
+        | ["games", "scripts", scriptId] => <ScriptDetails scriptId />
+        | _ => <NotFoundPage />
+        }
+      : <PageRestricted />;
+
+  <Language.Provider value=language>
+    <ReactIntlProvider locale={language.locale} messages={language.translations}>
+      <Notifications.Provider>
```

Comment:
`Language.Provider`, `ReactIntlProvider` and `Notifications.Provider` are not part of router mechanism so they should be in `App.re`, not `AppRouter.re`. Rest can stay in `AppRouter.re`.

Reply:
What about `NavigationBar`? I will move it also outside `AppRouter.re` to avoid passing `setLanguage` as an argument.

Reply:
ok